### PR TITLE
Issue/316 validate pf

### PR DESCRIPTION
**Pull request by AntoineRondelet**

### CODE REVIEW

File: ttnctl/util/payload_functions.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+func ValidatePayload(ctx log.Interface, code, payloadType string) (string, error) {
+	// We first parse the function to return if the AST cannnot be built (syntax error)
+	ctx.Info("Parsing function...")
+	err := parse.PayloadFunction(code)
+	if err != nil {
+		return "", err
+	}
+	ctx.Info("Function parsed successfully: syntax checked")
+
+	fmt.Printf("\n Test the function to detect runtime errors \n")
```
Comment (htdvisser) [inline ttnctl/util/payload_functions.go]: Allow skipping this

File: ttnctl/util/payload_functions.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	fmt.Println("Provide the signature of the payload function with test values")
```
Comment (htdvisser) [inline ttnctl/util/payload_functions.go]: This is not clear

File: ttnctl/util/payload_functions.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	fmt.Println(`
+Note:
+1) Use single quotes for strings: E.g: 'this is a valid string'
```
Comment (htdvisser) [inline ttnctl/util/payload_functions.go]: why?

File: ttnctl/util/payload_functions.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+2) Use the built-in function JSON.stringify() to provide json objects parameters: E.g: JSON.stingify({ valid: argument })
```
Comment (htdvisser) [inline ttnctl/util/payload_functions.go]: Just give an example depending on the function type, this is not very user friendly

File: ttnctl/util/payload_functions.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	testSignature := ReadFunction(ctx)
+
+	// If no syntax error, we run the function
+	ctx.Info("Testing...")
+	err = testPayload(ctx, code, testSignature, payloadType)
+	if err != nil {
+		return "", err
+	}
+
+	ctx.Info("The test is successful, the given function is a valid payload")
```
Comment (htdvisser) [inline ttnctl/util/payload_functions.go]: > the given function is a valid payload

wut?

File: ttnctl/util/payload_functions.go (hunk 1 above)
Comment (AntoineRondelet) [inline ttnctl/util/payload_functions.go]: Do you mean the following test or just the logs ?

Comment (htdvisser) [inline ttnctl/util/payload_functions.go]: You should be able to skip the runtime test

**Comment by AntoineRondelet** [comment]
I will reopen this PR later and test the functions handler side

**Pull request closed without merging**